woocommerce/woocommerce
# Filters: Short-circuit Variation calculations

Relevant issue: #24659 (does not close)

Added short circuit filter for get_available_variations.
Added skip filter for Variation_Data_Store->read_price_data.

Note: this PR is incomplete.

### All Submissions:






### Changes proposed in this Pull Request:

Added filters allowing third parties to short-circuit some very expensive variation calculations. Background tasks can then be used to generate this data over time for eventual consistency.

### How to test the changes in this Pull Request:

1. Generate thousands of variations on a product
2. Try to use the store or admin regarding this variable product, the site will not be responsive.
3. Add a filter of each type short circuiting expensive calculations.
4. Try to use the store or admin, the site should be responsive now, though there won't be "good" data.
5. Run a background task or something to generate said data in the background.

### Other information:


I'm not certain new tests will apply here, since they would just test `add_filters()`.



### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

```
Added short circuit filter for get_available_variations.
Added skip filter for Variation_Data_Store->read_price_data.
```
